### PR TITLE
chore: add specs for conditional code generation

### PR DESCRIPTION
**Pull request by katcipis**
Initial specs for conditional code generation on generate_file and generate_hcl.

### CODE REVIEW

File: docs/codegen/generate-file.md [hunk 1]
```diff
@@ -64,3 +64,24 @@ or even the project root, which then generates the file for multiple stacks.
...
 in failure of the overall code generation process.
+
+
+## Conditional Code Generation
+
+Conditional code generation is achieved by the use of the `condition` attribute.
+The `condition` attribute should always evaluate to a boolean. If `condition` 
+is true, the code will be generated for the given stack. If `condition` is false,
+the code will **not** be generated for the given stack.
+
```
Comment (i4ki) [inline docs/codegen/generate-file.md:75]: maybe makes sense to state that if `condition` is not provided, then the code is always generated.. but not sure, it seems clear enough.

**Review by i4ki — DISMISSED** [review]
lgtm

File: docs/codegen/generate-hcl.md [hunk 2]
```diff
@@ -125,6 +125,31 @@ Blocks defined at different levels with the same label aren't allowed, resulting
...
+  condition = tm_length(global.list) > 0
+  content {
+    locals {
+      list = global.list
+    }
+  }
+}
+```
+
+Will only generate code for stacks that define the global `generate_file` to true.
```
Comment (mariux) [inline docs/codegen/generate-hcl.md:150]: i guess this is a mismatch of whatthe condition actually says?

**Review by mariux — CHANGES_REQUESTED** [review]

File: docs/codegen/generate-hcl.md (hunk 2 above)
Comment (i4ki) [reply docs/codegen/generate-hcl.md:150]: I think this is leftover of a change I suggested. @katcipis need to fix here.

**Pull request merged**